ui/src/pages/Login/Login.tsx

```diff
@@ -4,29 +4,37 @@ import Form from "react-bootstrap/Form";
 import { BiLogInCircle } from "react-icons/bi";
 import { Formik, Form as FormikForm } from "formik";
 import * as Yup from "yup";
-import { userService } from "services";
 import { UserContext } from "contexts";
 import { TwoColumnHero, FormikInput, FeedbackButton } from "components";
-import { useMutation } from "react-query";
+import { usePostUserLogin } from "../../api/apiComponents";
 
 const validationSchema = Yup.object({
   loginOrEmail: Yup.string().required("Required"),
   password: Yup.string().required("Required"),
 });
 
-type LoginParams = Yup.InferType<typeof validationSchema>;
+export type LoginParams = Yup.InferType<typeof validationSchema>;
 
-export const Login: React.FC = () => {
+type Props = {};
+
+export const Login: React.FC<Props> = ({}) => {
   const {
     dispatch,
     state: { loggedIn },
   } = useContext(UserContext);
 
   const navigate = useNavigate();
 
-  const mutation = useMutation(userService.login, {
-    onSuccess: ({ apiKey }) => dispatch({ type: "SET_API_KEY", apiKey }),
-  });
+  const mutation = usePostUserLogin();
+  const { isSuccess, data } = mutation;
+  const login = mutation.mutateAsync;
+
+  useEffect(() => {
+    if (isSuccess) {
```

Comment:
Could it be done still with `onSuccess` option, but passed to `usePostUserLogin` hook?